frontend/viewer/src/lib/layout/Scotty.svelte

```diff
@@ -0,0 +1,43 @@
+<script lang="ts" context="module">
+  const portals = ['right-toolbar', 'app-bar-menu'] as const;
+  type ScottyPortalTarget = typeof portals[number];
+
+  export function initScottyPortalContext(): void {
+    portals.forEach((name) => {
+      setContext(name, writable<HTMLElement>());
```

Comment:
this should probably just be one context with a list of stores, or a single store with a list of elements, I'm not sure which is best

Reply:
I refactored it to only use 1 context item